Doc/whatsnew/3.11.rst

```diff
@@ -1407,58 +1407,125 @@ contributors are volunteers from the community.
 CPython bytecode changes
 ========================
 
-* The bytecode now contains inline cache entries, which take the form of
-  :opcode:`CACHE` instructions. Many opcodes expect to be followed by an exact
-  number of caches, and instruct the interpreter to skip over them at runtime.
-  Populated caches can look like arbitrary instructions, so great care should be
-  taken when reading or modifying raw, adaptive bytecode containing quickened
-  data.
 
-* Replaced all numeric ``BINARY_*`` and ``INPLACE_*`` instructions with a single
-  :opcode:`BINARY_OP` implementation.
 
-* Replaced the three call instructions: :opcode:`CALL_FUNCTION`,
-  :opcode:`CALL_FUNCTION_KW` and :opcode:`CALL_METHOD` with
-  :opcode:`PUSH_NULL`, :opcode:`PRECALL`, :opcode:`CALL`,
-  and :opcode:`KW_NAMES`.
-  This decouples the argument shifting for methods from the handling of
-  keyword arguments and allows better specialization of calls.
-
-* Removed ``COPY_DICT_WITHOUT_KEYS`` and ``GEN_START``.
-
-* :opcode:`MATCH_CLASS` and :opcode:`MATCH_KEYS` no longer push an additional
-  boolean value indicating whether the match succeeded or failed. Instead, they
-  indicate failure with :const:`None` (where a tuple of extracted values would
-  otherwise be).
-
-* Replace several stack manipulation instructions (``DUP_TOP``, ``DUP_TOP_TWO``,
-  ``ROT_TWO``, ``ROT_THREE``, ``ROT_FOUR``, and ``ROT_N``) with new
-  :opcode:`COPY` and :opcode:`SWAP` instructions.
-
-* Replaced :opcode:`JUMP_IF_NOT_EXC_MATCH` by :opcode:`CHECK_EXC_MATCH` which
-  performs the check but does not jump.
-
-* Replaced :opcode:`JUMP_IF_NOT_EG_MATCH` by :opcode:`CHECK_EG_MATCH` which
-  performs the check but does not jump.
-
-* Replaced :opcode:`JUMP_ABSOLUTE` by the relative :opcode:`JUMP_BACKWARD`.
-
-* Added :opcode:`JUMP_BACKWARD_NO_INTERRUPT`, which is used in certain loops where it
-  is undesirable to handle interrupts.
-
-* Replaced :opcode:`POP_JUMP_IF_TRUE` and :opcode:`POP_JUMP_IF_FALSE` by
-  the relative :opcode:`POP_JUMP_FORWARD_IF_TRUE`, :opcode:`POP_JUMP_BACKWARD_IF_TRUE`,
-  :opcode:`POP_JUMP_FORWARD_IF_FALSE` and :opcode:`POP_JUMP_BACKWARD_IF_FALSE`.
-
-* Added :opcode:`POP_JUMP_FORWARD_IF_NOT_NONE`, :opcode:`POP_JUMP_BACKWARD_IF_NOT_NONE`,
-  :opcode:`POP_JUMP_FORWARD_IF_NONE` and :opcode:`POP_JUMP_BACKWARD_IF_NONE`
-  opcodes to speed up conditional jumps.
-
-* :opcode:`JUMP_IF_TRUE_OR_POP` and :opcode:`JUMP_IF_FALSE_OR_POP` are now
-  relative rather than absolute.
+.. _whatsnew311-added-opcodes:
```

Comment:
The list we have now has all the jump related changes together, but if you split it into a "added" table and a "replaced" table then they are no longer together. I don't know if that's an improvement.

Reply:
I created an alternative PR -https://github.com/python/cpython/pull/98566.   It tries to consolidate related items into a high level descriptions, rather than enumerate all the changes. Maybe this is more useful for whatsnew. The full details, if you need them, as in the dis doc.

I added an item about the exc-info related changes. Didn't do anything about these:

GET_AWAITABLE

WITH_EXCEPT_START
If the low bit of namei is set, then a NULL is pushed to the stack before the global variable.

LOAD_GLOBAL
i is no longer offset by the length of co_varnames

LOAD_CLOSURE
LOAD_DEREF
LOAD_CLASSDEREF
STORE_DEREF
DELETE_DEREF

Reply:
Great points, thanks. I've incorporated the changes suggested via your PR, specifically:

* Consolidated the jump opcode additions, replacements and changes (except for the ``NO_INTERRUPT`` opcode addition, which didn't appear to be closely related to the others) into a single Replaced table entry with a linked note, and also just linked the high-level jump opcode name patterns, instead of every opcode.
* Added a "Changed" note about the exception stack representation
* `EXC`/`EG` changes were already consolidated

Beyond that, I've given this PR a major revamp based on your feedback above:

* Converted the "Added" section back to (much shorter) bullets
* Inlined the previous separate notes into the Replaced table
* Combined the previous "Removed" and "Changed" sections into one, condensing the Removed items into a single bullet to take up much less space
* Made additional textual refinements, particularly in the Changed entries, to be both more clear and more concise

In total, this retains nearly the same space consumption of the existing flat-bullets section (once allowing for the missing substantive additions, replacements and removals added here), while being much easier for the reader to quickly navigate, consult and jump to more information in the `dis` docs than the previous wall of bullets.

| Flat Bullets  | Initial PR | Revised PR |
| ------------- | ------------- | ---- |
| ![image](https://user-images.githubusercontent.com/17051931/197420407-25849f59-939c-4a82-a4ff-4a094ddaac41.png) | ![image](https://user-images.githubusercontent.com/17051931/197363352-04d4f23e-d026-4b02-afe5-01ca83ddd6b5.png)  | ![image](https://user-images.githubusercontent.com/17051931/197420114-a212be6c-cc4b-485d-b8d9-bafd16658bbd.png) |